book/style_guide.md

```diff
@@ -0,0 +1,221 @@
+
+### Basic
+
+- **It's recommended to** put one space before and after pipe `|` symbol, commands, subcommands, their options and arguments.
+- **It's recommended to** omit commas between list items.
+
+Correct:
+
+```nushell
+'Hello, Nushell! This is a gradient.' | ansi gradient --fgstart '0x40c9ff' --fgend '0xe81cff'
+```
+
+Incorrect:
+
+```nushell
+'Hello, Nushell! This is a gradient.' |  ansi gradient --fgstart '0x40c9ff' --fgend '0xe81cff' # two many spaces after "|": 2 instead of 1
+```
+
+#### One-line format
+
+One-line format is a format for writing all commands in one line.
+
+1. parameters:
+   1. **It's recommended to** put no spaces before and after pipe `|` symbol denoting block or closure parameter list beginning.
+   2. **It's recommended to** put one space after comma `,` after block or closure parameter parameter.
+   3. **It's recommended to** put no space before and one space after pipe `|` symbol denoting block or closure parameter list end.
+2. block and closure bodies:
+   1. **It's recommended to** put one space before closing block or closure curly brace `}`.
+3. records:
+   1. **It's recommended to** put one space before first record key and after last record key value.
+   2. **It's recommended to** put one space after `:` after record key.
+   3. **It's recommended to** put one space after comma `,` after key value.
+4. lists:
+   1. **It's recommended to** put no spaces before first list value and after last list value.
+   2. **It's recommended to** put one space after comma `,` after list value.
+5. surrounding constructs:
+   1. **It's recommended to** put one space before opening square `[` or curly brace `{` if preceding symbol is not the same.
+   2. **It's recommended to** put one space after closing square `]` or curly brace `}` if following symbol is not the same.
+   3. **It's recommended to** put no spaces between square `[]` or curly brackets `{}` with nothing between them.
+
+Correct:
+
+```nushell
+[[status]; [UP] [UP]] | all {|el| $el.status == UP }
+[1 2 3 4] | reduce {|it, acc| $it + $acc }
+{ x: 1, y: 2 }
+[1 2] | zip [3 4]
+[]
+```
+
+Incorrect:
+
+```nushell
+[[status]; [UP] [UP]] | all { |el| $el.status == UP } # two many spaces before "|el|": 2 instead of 0
+[1 2 3 4] | reduce {|it , acc | $it + $acc } # two many spaces after "it": 1 instead of 0
+                                             # two many spaces before "| $it + $acc": 1 instead of 0
+{x: 1, y : 2 } # two few spaces before "x: 1": 0 instead of 1
+               # two many spaces before ": 2": 1 instead of 0
+[1 2] | zip  [3 4] # two many spaces before "[3 4]": 2 instead of 1
+[ ] # two many spaces before "]": 1 instead of 0
+```
+
+#### Multi-line format
+
+Multi-line format is a format for writing all commands in several lines. It inherits all rules from one-line format
+and modifies them slightly.
+
+1. general:
+   1. **It's recommended to omit** trailing spaces.
```

Comment:
i think this one should be mandatory actually :scream: :laughing: 
(i hate uncontrolled and unwanted trailing whitespaces :smirk:)

Reply:
I guess formatter we gonna create should automatically remove them. ;)

Reply:
yeeeeah of course, but still quite a plague these trailing and sometimes invisible characters :sob:

Reply:
I've changed this rule to requirenment.